cairo-contracts/packages/apps/src/transfer/components/transfer.cairo

```diff
@@ -277,6 +277,29 @@ pub mod TokenTransferComponent {
         }
     }
 
+    // -----------------------------------------------------------
+    // ICreateToken
+    // -----------------------------------------------------------
+
+    #[embeddable_as(CreateIbcToken)]
+    impl CreateTokenImpl<
+        TContractState, +HasComponent<TContractState>, +Drop<TContractState>,
+    > of ICreateIbcToken<ComponentState<TContractState>> {
+        fn create_ibc_token(
+            ref self: ComponentState<TContractState>,
+            chan_id_on_b: ChannelId,
+            base_denom: ByteArray,
+        ) -> ContractAddress {
+            let token = self.create_token(base_denom.clone(), 0);
+            let trace_prefix = TracePrefixTrait::new(TRANSFER_PORT_ID(), chan_id_on_b);
```

Comment:
Also, in the current form of `create_ibc_token`, Osmosis can only index tokens native to itself -- `OSMO` and `ION`. The remote IBC tokens, such as, `ATOM` (on Osmosis) can't be created currently.

Reply:
Good point. Let me see what can be done for this.